presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -196,6 +210,11 @@ private Expression unwrapCast(ComparisonExpression expression)
             Type sourceType = typeAnalyzer.getType(session, types, cast.getExpression());
             Type targetType = typeAnalyzer.getType(session, types, expression.getRight());
 
+            if (targetType instanceof TimestampWithTimeZoneType) {
+                // Note: two TIMESTAMP WITH TIME ZONE values differing in zone only are considered equal.
```

Comment:
If this is something the SQL spec says then I would like if the comment ended with `must be considered equal`.

From what I can see the spec says equality is when the UTC Instants are same (regardless of offset used).

Reply:
> Indeed, that comment is incorrect. They are considered equal if their instant in the same.

This is what i meant, i added "same instant"  to make the comment clearer
```
// Note: two TIMESTAMP WITH TIME ZONE values differing in zone only (same instant) are considered equal.
```

> This deserves a better comment, too. The reason we reinterpret the value in the session timezone is so that the conversion roundtrips properly when going from timestamp w/ tz -> timestamp -> timestamp w/ tz.

Yes, this is the point. I put the check & explanation in `io.prestosql.sql.planner.iterative.rule.UnwrapCastInComparison.Visitor#hasInjectiveImplicitCoercion` method.